src/material-experimental/mdc-list/list-option.html

```diff
@@ -29,24 +30,29 @@
   </div>
 </ng-template>
 
-<!-- Prefix -->
-<span class="mdc-list-item__graphic"
-      *ngIf="!_isReversed() && (_hasIconOrAvatar() || _hasCheckbox())">
-  <ng-container [ngTemplateOutlet]="_hasCheckbox() ? checkbox : icons">
-  </ng-container>
+<!-- Container for the checkbox at start. -->
+<span class="mdc-list-item__graphic mat-mdc-list-option-checkbox-before"
+      *ngIf="_shouldShowCheckboxAt('before')">
+  <ng-template [ngTemplateOutlet]="checkbox"></ng-template>
 </span>
+<!-- Conditionally renders icons/avatars before the list item text. -->
+<ng-template [ngIf]="_shouldShowIconsAndAvatarsAt('before')">
+  <ng-template [ngTemplateOutlet]="icons"></ng-template>
+</ng-template>
 
 <!-- Text -->
 <span class="mdc-list-item__text" #text [id]="_optionTextId">
   <ng-content></ng-content>
 </span>
 
-<!-- Suffix -->
-<span class="mdc-list-item__meta"
-      *ngIf="_isReversed() && (_hasCheckbox() || _hasIconOrAvatar())">
-  <ng-container [ngTemplateOutlet]="_hasCheckbox() ? checkbox : icons">
-  </ng-container>
+<!-- Container for the checkbox at the end. -->
+<span class="mdc-list-item__meta" *ngIf="_shouldShowCheckboxAt('after')">
+  <ng-template [ngTemplateOutlet]="checkbox"></ng-template>
 </span>
+<!-- Conditionally renders icons/avatars after the list item text. -->
+<ng-template [ngIf]="_shouldShowIconsAndAvatarsAt('after')">
+  <ng-template [ngTemplateOutlet]="icons"></ng-template>
```

Comment:
I think you can drop the `ngIf` and the `ng-template` here like this:

```
<ng-container [ngTemplateOutlet]="_shouldShowIconsAndAvatarsAt('after') ? icons : null"></ng-container>
```

Same goes for the other outlets.

Reply:
Looks like this doesn't work and when the template outlet conditional changes, one outlet will not render anything. It might be related to timing of the embedded view that is shared between two outlets then. With a separate template it might avoid this problematic timing due to `ngIf` first being resolved and then the outlet being rendered afterwards.